tests/range/range_api.cpp

```diff
@@ -254,6 +262,14 @@ class TEST_NAME : public util::test_base {
         test2d(log, range_2d_g, range_2d_l, my_queue);
         test_range<3> test3d;
         test3d(log, range_3d_g, range_3d_l, my_queue);
+
+        // 32 bits, it's trivial. Sanity test.
```

Comment:
sizeof(unsigned int) is not guaranteed to be 32 bits. I suggest using unit32_t.

Reply:
Good call. The doc C++ seems to imply that `std::numeric_limits` didn't took `unit32_t` but I just checked and it worked. So will switch to that.